### ISSUE TITLE
Filter Class for other openGL versions

### ISSUE THREAD

**Issue by TheMagnat** [glumpy/glumpy#272]
Hello,

I was trying to use the Filter Class with GL 3.3 but I discovered that the class was only adapted for old GLSL.
How about adding options to use this class with more recent version of GL ? (Or creating another class I don't know)

For the moment I just copied the class and edited the part that clash with 3.3 GL to make it work.
I could share this little edit in a pull request but I'm not sure if it's working fine in every situations (looks good at the moment) and if I should create a new class or add option to the Filter class.

What do you think of this feature ?

**Comment by rougier** [comment]
I target GLES 2.0 and this explains the old GLSL code. Problem is that with modern GLSL, I would need to rerite part of the parser engine to cope with new keywords and injection point. For the filter, you mean the code is not working at all with GL 3.3?

**Comment by TheMagnat** [comment]
Yes, since it use constant code for its shaders (wrote in old GLSL) and it also need to use vertex array.
I also found that the alpha channel is ignored in this filter class, I'm trying to make it work (So I could draw something, draw something else in front of it with only this thing getting filtered)
